cmake/VcpkgToolchain.cmake

```diff
@@ -90,3 +89,6 @@ set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
 
 # Copies DLLs built by vcpkg when an install() command is run. Probably only works on Windows.
 set(X_VCPKG_APPLOCAL_DEPS_INSTALL TRUE CACHE BOOL "")
+
+# Target 10.14 when building for macOS
+set(VCPKG_OSX_DEPLOYMENT_TARGET "10.14")
```

Comment:
[This](https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_osx_deployment_target) setting only works in triplets. The analogous CMake setting for our project is [CMAKE_OSX_DEPLOYMENT_TARGETS](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html). It probably needs to be set in the main CMakeLists.txt file while `VCPKG_OSX_DEPLOYMENT_TARGET` needs to be moved back into our osx-plasma triplets.

Reply:
oh, you can't specify this as `cmake -DVCPKG_OSX_DEPLOYMENT_TARGET=10.14` on the command-line? No wonder our logs are full of weird "libraries built for a version newer than your target" warnings 🤦🏼‍♂️

Reply:
I can reimplement the triplets - but I'd need some guidance first. The triplets don't seem to follow the standard formatting expected by vcpkg - and the Cmake seems to force the non standard Windows triplets to load. Is there a reason we should force the specific triplet to load in CMake? Or can I just name the triplet the same as the built in triplet and let vcpkg apply the overlay itself?

Reply:
See https://github.com/H-uru/Plasma/pull/1444#issuecomment-1661117834 for the answer to most of this question. The short version of it is that we have an opinion about how our build should work. The official vcpkg Windows triplets work perfectly but don't match our opinion. You have the same situation on macOS - the official triplet will build and work fine, but we want to set a certain preference for the deployment target out of the box. So, we provide that in an overlay triplet.

> the Cmake seems to force the non standard Windows triplets to load

Just to be clear, we set a default triplet in VcpkgToolchain.cmake, but whoever is doing the build is free to change the `VCPKG_TARGET_TRIPLET` if they want. We're opinionated but not dictatorial.

> The triplets don't seem to follow the standard formatting expected by vcpkg

I don't understand what you mean here.

Reply:
As discussed on IRC - I'm happy to let @dpogue deal with this as part of his pull request. But I did want to follow up on my question here just for my own knowledge.

My understanding of the triplets is they should be named something like "x86-windows.cmake" to override the built in triplet (which has the same name.) But ours are named "x86-windows-plasma.cmake" and then manually loaded in through CMake. Is there a reason we use the plasma suffix? (Also totally possible I'm not understanding the triplet system fully.)

Reply:
@colincornaby See Hoikas's comment here: https://github.com/H-uru/Plasma/pull/1444#issuecomment-1661117834

Reply:
Ah - ok. I should have just clicked the link the first time when @Hoikas posted it!